File: PersonalProjects/prosjekt-3/backend/crud/director.ts
import { Client, QueryResult, QueryResultRow } from "pg";
import { Director } from "../types";

const getExtras = async (result: QueryResult) => {
  for (let i = 0; i < result.rows.length; i++) {
    const row = result.rows[i];
    row.image = process.env.STATIC_ENDPOINT + `/static/directors/${row.id}.jpg`;
  }
  return result.rows;
};

export const getDirectors = async (
  row: QueryResultRow,
  client: Client
): Promise<Director[]> => {
  const result = await client.query(
    "select d.id, d.name from moviedirector md inner join director d on d.id = md.directorid  where md.movieid = $1::uuid",
    [row.id]
  );
  const rows = getExtras(result);
  return rows;
};

File: PersonalProjects/prosjekt-3/data/scraper/director.py

from uuid import uuid4

class Director:

    directors = {}

    def __init__(self):
        pass   

    def getDirector(self, director):
        d = director.split('(')[0]
        if (d in Director.directors.keys()):
            return Director.directors[d]
        else:
            id = uuid4()
            Director.directors[d] = {'id': id, 'name': d}
            return Director.directors[d]

    def getDirectors(self, director):
        return [Director.getDirector(self, d.strip()) for d in director.split(",")]